s3transfer/utils.py

```diff
@@ -300,7 +302,12 @@ def is_special_file(cls, filename):
         return False
 
     def get_temp_filename(self, filename):
-        return filename + os.extsep + random_file_extension()
+        sep = os.extsep
+        ext = random_file_extension()
+        path = os.path.dirname(filename)
+        name = os.path.basename(filename)
+        temp_filename = name[:self._MAX_FILENAME_LEN - len(sep) - len(ext)] + sep + ext 
```

Comment:
This line is getting a little long, let's split it up so it's under 80 characters. Considering we always use `sep` and `ext` together it might make sense to just combine them into a `suffix` like:
```
suffix = os.extsep + random_file_extension()
...
temp_filename = name[:self._MAX_FILENAME_LEN - len(suffix)] + suffix
```